Automattic/wp-calypso
# Onboarding - Add a new flow for readymade templates

Adds a new onboarding flow for the readymade templates, including the Launchpad integration.

Related to #https://github.com/Automattic/dotcom-forge/issues/7814 and https://github.com/Automattic/dotcom-forge/issues/7815

## Proposed Changes
Adds a new onboarding flow for Readymade templates that:
* Sets the Assembler theme on the site
* Applies the Readymade template
* Redirects the user to the Site Editor
* When the user exists the SE, the launchpad is triggered for this specific flow

## Why are these changes being made?


* Needed in the context of Readymade templates in Patterns Library.

## Testing Instructions



* Apply the WPCOM changes from [this](https://github.com/Automattic/jetpack/pull/37989) PR to your sandbox: `bin/jetpack-downloader test jetpack-mu-wpcom-plugin add/readymade-template-launchpad-task-list`
* Use the live link and go to `/setup/readymade-template?readymadeTemplateId=1`
* This will automatically create a site for you
* Will now land in the site-editor
* Exit the Site Editor and notice that you land in the launchpad
* Notice that the launchpad URL is `setup/readymade-template/launchpad`
* Click on the domains tab
* You should be able to change your WordPress.com subdomain besides registering a new domain
* Try out other tabs and see if they are working fine (except for the Write post that sends you to production launchpad - causing a redirect since the flow doesn't exist there)

## Pre-merge Checklist
